include/zephyr/arch/riscv/csr.h

```diff
@@ -233,4 +234,40 @@
 				: "memory");			\
 })
 
+/* uhartid -- user hardware thread id, seems to be not yet defined in toolchain */
+#define	uarthid		0x014
+
+#ifdef CONFIG_RISCV_KERNEL_IN_USER_MODE
+	/* zephyr runs in user mode */
+
+	/* inline assembly defines */
+	#define XSTATUS_IEN			USTATUS_IEN
+	#define XSTATUS_DEF_RESTORE		USTATUS_DEF_RESTORE
+	#define XSTATUS				0x000
+
+	/* register definition for assembly */
+	#define xhartid				uarthid
```

Comment:
nit: should be `uhartid`

Reply:
typo: this is the [line](https://github.com/zephyrproject-rtos/zephyr/blob/e053ab2b382f846d584db3d8ff18f5816e574de7/include/zephyr/arch/riscv/csr.h#L238) that fixes it :D 

this is self defined as this is not built-in in the riscv gcc.